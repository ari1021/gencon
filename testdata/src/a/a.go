package a

func e[T any](t T) {} // want "change constraint of T from any to string|~int"

func f[T, U any, V int](t T, u U, v V) {} // want "change constraint of T from any to bool|int" "change constraint of U from any to string|~int"

func g(s string) {} // OK

func invoke() { // OK
	e("gopher")
	e(1)
	type MyInt int
	e(MyInt(18))

	f(3, "gopher", 100)
	f(true, MyInt(3), 100)

	g("gopher")
}
