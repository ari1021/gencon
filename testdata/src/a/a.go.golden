package a

func e[T string | ~int](t T) {} // want "should not use 'any'. hint: string|~int"

func f[T, U any, V int](t T, u U, v V) {} // want "should not use 'any'. hint: bool|int" "should not use 'any'. hint: string|~int"

func g(s string) {} // OK

func h[T any](t T) {} // want "should not use 'any'"

func invoke() { // OK
	e("gopher")
	e(1)
	type MyInt int
	e(MyInt(18))

	f(3, "gopher", 100)
	f(true, MyInt(3), 100)

	g("gopher")
}
